frontend/src/components/common/ActionMenuItem/ActionMenuItem.tsx

```diff
@@ -0,0 +1,43 @@
+import { Icon, IconifyIcon, IconProps } from '@iconify/react';
+import { ListItemIcon, ListItemText, MenuItem } from '@mui/material';
+import React from 'react';
+
+export interface ActionMenuItemProps {
+  /** A short description of the action. */
+  description: string;
+  /** Either a string icon, or imported icon. */
+  icon: string | IconifyIcon;
```

Comment:
You could just type this as `Icon['icon']` or `IconifyIcon` solely, in case the latter includes string there.